dashboard/src/components/__tests__/TimeDropdown.test.ts

```diff
@@ -1,114 +1,127 @@
     const date = new Date(Date.UTC(2025, 0, 1, 12, 0, 0));
     vi.setSystemTime(date);
 
-    const wrapper = mount(TimeDropdown, {
-      props: {
-        fieldname: "testField",
-        onChange: handleChange,
-      },
-    });
+    await wrapper.find("select").setValue(value);
 
-    const options = wrapper.findAll(".dropdown-item");
-    await options[index].trigger("click");
-    expect(changeEvent.value).toEqual({
-      field: "testField",
-      value: want,
-    });
+    expect(wrapper.emitted("change")).toEqual([["createdAt", want, ""]]);
   });
 
   it.each([
-    [0, "Started"], // Any time.
-    [1, "Started: The last 3 hours"],
-    [2, "Started: The last 6 hours"],
-    [3, "Started: The last 12 hours"],
-    [4, "Started: The last 24 hours"],
-    [5, "Started: The last 3 days"],
-    [6, "Started: The last 7 days"],
+    ["3h", "Started: The last 3 hours"],
+    ["6h", "Started: The last 6 hours"],
+    ["12h", "Started: The last 12 hours"],
+    ["24h", "Started: The last 24 hours"],
+    ["3d", "Started: The last 3 days"],
+    ["7d", "Started: The last 7 days"],
   ])(
     "sets the button label correctly when a time is selected",
-    async (index, want) => {
-      const wrapper = mount(TimeDropdown, {
-        props: {
-          fieldname: "testField",
-          onChange: handleChange,
-        },
-      });
-
+    async (value, want) => {
       const button = wrapper.find("button");
-      const option = wrapper.findAll(".dropdown-item");
 
-      await option[index].trigger("click");
+      await wrapper.find("select").setValue(value);
+
       expect(button.text()).toBe(want);
     },
   );
 
-  it("resets the button label when 'Any time' is selected", async () => {
-    const wrapper = mount(TimeDropdown, {
-      props: {
-        fieldname: "testField",
-        onChange: handleChange,
-      },
-    });
-
+  it("clears all values when the clear button is clicked", async () => {
     const button = wrapper.find("button");
-    const option = wrapper.findAll(".dropdown-item");
+    const clearButton = wrapper.find("button[type='reset']");
 
-    await option[1].trigger("click");
-    expect(button.text()).toBe("Started: The last 3 hours");
+    await clearButton.trigger("click");
 
-    await option[0].trigger("click");
     expect(button.text()).toBe("Started");
+    expect(wrapper.emitted("change")).toEqual([["createdAt", "", ""]]);
+  });
+
+  it("emits the correct event when a custom date is selected", async () => {
+    const button = wrapper.find("button");
+    const datePicker = wrapper.findComponent<typeof VueDatePicker>(
+      '[data-test="startTime"]',
+    );
+
+    datePicker.vm.$emit(
+      "update:model-value",
+      new Date(Date.UTC(2025, 0, 1, 12, 0, 0)),
+    );
+    await datePicker.vm.$nextTick();
+
+    expect(button.text()).toBe("Started: Custom");
+    expect(wrapper.emitted("change")).toEqual([
+      ["createdAt", "2025-01-01T12:00:00Z", ""],
+    ]);
   });
```

Comment:
This test is making a previous one ("handles direct date picker input") unnecessary.

Reply:
I kept this one that also checks the button and removed "handles direct date picker input".